src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp

```diff
@@ -246,34 +248,33 @@ class KerrSchild : public MarkAsAnalyticSolution {
 
   template <typename DataType>
   using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataType>,
-                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+                                   tmpl::size_t<volume_dim>, Frame>;
   template <typename DataType>
   using DerivShift =
-      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
-                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame, DataType>,
+                    tmpl::size_t<volume_dim>, Frame>;
   template <typename DataType>
   using DerivSpatialMetric = ::Tags::deriv<
-      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
-      tmpl::size_t<volume_dim>, Frame::Inertial>;
+      gr::Tags::SpatialMetric<volume_dim, Frame, DataType>,
+      tmpl::size_t<volume_dim>, Frame>;
   template <typename DataType>
   using tags = tmpl::list<
       gr::Tags::Lapse<DataType>, ::Tags::dt<gr::Tags::Lapse<DataType>>,
       DerivLapse<DataType>,
-      gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>,
-      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataType>>,
+      gr::Tags::Shift<volume_dim, Frame, DataType>,
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame, DataType>>,
       DerivShift<DataType>,
-      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>,
+      gr::Tags::SpatialMetric<volume_dim, Frame, DataType>,
       ::Tags::dt<
-          gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataType>>,
+          gr::Tags::SpatialMetric<volume_dim, Frame, DataType>>,
       DerivSpatialMetric<DataType>, gr::Tags::SqrtDetSpatialMetric<DataType>,
-      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataType>,
-      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataType>>;
+      gr::Tags::ExtrinsicCurvature<volume_dim, Frame, DataType>,
+      gr::Tags::InverseSpatialMetric<volume_dim, Frame, DataType>>;
 
   template <typename DataType, typename... Tags>
-  tuples::TaggedTuple<Tags...> variables(const tnsr::I<DataType, volume_dim>& x,
-                                         double /*t*/,
-                                         tmpl::list<Tags...> /*meta*/) const
-      noexcept {
+  tuples::TaggedTuple<Tags...> variables(
```

Comment:
I believe you only need to template this function on `Frame`, plus all the things that are currently templated on `DataType` in this class. Then the public interface can remain unchanged. Do you think that'll work?

Reply:
Yes, this works!  The changes to KerrSchild are more complicated, but the only things that need to change are KerrSchild and the test.